can/broadcastmanager.py

```diff
@@ -34,28 +36,86 @@ class CyclicSendTaskABC(CyclicTask):
     Message send task with defined period
     """
 
-    def __init__(self, message, period):
+    def __init__(self, messages, period):
         """
-        :param can.Message message: The message to be sent periodically.
-        :param float period: The rate in seconds at which to send the message.
+        :param Union[List[can.Message], tuple(can.Message), can.Message] messages:
+            The messages to be sent periodically.
+        :param float period: The rate in seconds at which to send the messages.
         """
-        self.message = message
-        self.can_id = message.arbitration_id
-        self.arbitration_id = message.arbitration_id
+        messages = self._check_and_convert_messages(messages)
+
+        # Take the Arbitration ID of the first element
+        self.arbitration_id = messages[0].arbitration_id
         self.period = period
-        super().__init__()
+        self.messages = messages
+
+    @staticmethod
+    def _check_and_convert_messages(messages):
+        """Helper function to convert a Message or Sequence of messages into a
+        tuple, and raises an error when the given value is invalid.
+
+        Performs error checking to ensure that all Messages have the same
+        arbitration ID and channel.
+
+        Should be called when the cyclic task is initialized
+        """
+        if not isinstance(messages, (list, tuple)):
+            if isinstance(messages, can.Message):
+                messages = [messages]
+            else:
+                raise ValueError("Must be either a list, tuple, or a Message")
+        if not messages:
+            raise ValueError("Must be at least a list or tuple of length 1")
+        messages = tuple(messages)
+
+        all_same_id = all(
+            message.arbitration_id == messages[0].arbitration_id for message in messages
+        )
+        if not all_same_id:
+            raise ValueError("All Arbitration IDs should be the same")
+
+        all_same_channel = all(
+            message.channel == messages[0].channel for message in messages
+        )
+        if not all_same_channel:
+            raise ValueError("All Channel IDs should be the same")
+
+        return messages
+
+    def _check_modified_messages(self, messages):
```

Comment:
Not sure this is the write location for this method since it's only ever use by `ModifiableCyclicTaskABC.modify_data`. Maybe it should be in the `ModifiableCyclicTaskABC` class or nested inside `ModifiableCyclicTaskABC.modify_data`.